lib/middleware.ts

```diff
@@ -0,0 +1,5 @@
+export { default } from 'next-auth/middleware';
+
+export const config = {
+  matcher: ['/api/:function*'],
+};
```

Comment:
can you add a jsdoc for why this middleware is necessary and what it does?

Reply:
Sure